src/auth/mas.py

```diff
@@ -2,32 +2,192 @@
 # See LICENSE file for licensing details.
 
 """Helper module used to manage MAS-related workloads."""
-
 import logging
+import secrets
+import typing
 
+import ops
 from jinja2 import Environment, FileSystemLoader, select_autoescape
 
+from charm_types import SMTPConfiguration
 from state.charm_state import SynapseConfig
 from state.mas import MASConfiguration
 
+logger = logging.getLogger()
+
 MAS_TEMPLATE_FILE_NAME = "mas_config.yaml.j2"
 
-logger = logging.getLogger()
+MAS_SERVICE_NAME = "synapse-mas"
+MAS_EXECUTABLE_PATH = "/usr/bin/mas-cli"
+MAS_WORKING_DIR = "/mas"
+MAS_CONFIGURATION_PATH = f"{MAS_WORKING_DIR}/config.yaml"
+
+MAS_PEBBLE_LAYER = ops.pebble.LayerDict(
+    {
+        "summary": "Matrix Authentication Service layer",
+        "description": "pebble config layer for MAS",
+        "services": {
+            MAS_SERVICE_NAME: {
+                "override": "replace",
+                "summary": "Matrix Authentication Service",
+                "startup": "enabled",
+                "command": f"{MAS_EXECUTABLE_PATH} server -c {MAS_CONFIGURATION_PATH}",
+                "working-dir": MAS_WORKING_DIR,
+            }
+        },
+    }
+)
+
+# Disabling it since this is only the label for juju secret
+ADMIN_TOKEN_SECRET_LABEL = "admin.token"  # nosec
+
+
+class MASConfigInvalidError(Exception):
+    """Exception raised when validation of the MAS config failed."""
+
+
+class MASRegisterUserFailedError(Exception):
+    """Exception raised when validation of the MAS config failed."""
+
+
+class MASVerifyUserEmailFailedError(Exception):
+    """Exception raised when validation of the MAS config failed."""
+
+
+class MASGenerateAdminAccessTokenError(Exception):
+    """Exception raised when generation of admin token failed."""
 
 
-# pylint: disable=too-few-public-methods
+def validate_mas_config(container: ops.model.Container) -> None:
+    """Validate current MAS configuration.
+
+    Args:
+        container: Synapse container.
+    """
+    command = [MAS_EXECUTABLE_PATH, "config", "check", "-c", MAS_CONFIGURATION_PATH]
+    process = container.exec(command=command, working_dir=MAS_WORKING_DIR)
```

Comment:
Should wrap the pebble exec error into a business error, same comment for `sync_mas_config`

Reply:
I updated the code to raise a custom exception here